nex/storage-manager/activate_with_card_id.go

```diff
@@ -0,0 +1,67 @@
+package nex_storage_manager
+
+import (
+	"database/sql"
+
+	"github.com/PretendoNetwork/mario-kart-7/database"
+	"github.com/PretendoNetwork/mario-kart-7/globals"
+	nex "github.com/PretendoNetwork/nex-go"
+	storage_manager "github.com/PretendoNetwork/nex-protocols-go/storage-manager"
+)
+
+func ActivateWithCardID(err error, client *nex.Client, callID uint32, unknown uint8, cardID uint64) uint32 {
+	if err != nil {
+		globals.Logger.Error(err.Error())
+		return nex.Errors.Core.InvalidArgument
+	}
+
+	// * It's not guaranteed that the client will call AcquireCardID,
+	// * because that method is only called the first time the client
+	// * goes online, and it may have used official servers previously.
+	// *
+	// * To workaround this, we ignore the card ID stuff and get the
+	// * unique ID using the PID
```

Comment:
Is this the case even between saves?

Reply:
It's probably not, the card ID is stored on the save

This is a workaround to allow users to transition to our servers without erasing their save. The only issue I see with keeping the save is that people will keep their old rivals list (the PIDs of people they have played online with before), and that could cause some "PID collisions", where they would see pretendo players connected as their nintendo equivalents (this is checked using `GetSimplePlayingSession`).

I'm not sure of the best way to go with this tbh

Reply:
My bad I wasn't clear, I meant the account saves. Nimbus creates a new user account, which gets it's own act/frd save file. I had assumed this, like on the WiiU, meant that this new user would *also* have a new save for the games on the console

Thinking about it though, if the 3DS game stores save data on the cart then that wouldn't make much sense since the same save is available on any console

I think what you've done is fine. It's a hack, but it's the best we can do imo without forcing a user to completely save wipe